lms/djangoapps/mobile_api/social_facebook/groups/views.py

```diff
@@ -0,0 +1,123 @@
+"""
+Views for groups info API
+"""
+
+from rest_framework import generics, status, mixins
+from rest_framework.response import Response
+from ...utils import mobile_view
+from lms.djangoapps.mobile_api.social_facebook.groups import serializers
+import facebook
+
+from django.conf import settings
+_FACEBOOK_API_VERSION = settings.FACEBOOK_API_VERSION
+_FACEBOOK_APP_ID = settings.FACEBOOK_APP_ID
+_FACEBOOK_APP_SECRET = settings.FACEBOOK_APP_SECRET
+
+@mobile_view()
+class Groups(generics.CreateAPIView, mixins.DestroyModelMixin):
+    """
+    **Use Case**
+
+        An API to Create or Delete course groups.
+
+    **Creation Example request**:
+
+        POST /api/mobile/v0.5/social/facebook/groups/<group_id>
```

Comment:
You're right, the app doesn't know which group id to send before it was created.
The reason this is there is because the create and delete are using the same API endpoint. 
The delete needs to know what group_id to delete but for the purpose of creation the group_id can be left empty or ignored. 
I can see how this can be confusing and it gives rise the issue that when the group_id is necessary the view can be invoked without the group_id. 
Any thoughts on best practices dealing with a case like this?